geode-core/src/main/java/org/apache/geode/internal/cache/TXStateProxyImpl.java

```diff
@@ -531,8 +591,13 @@ public Object findObject(KeyInfo key, LocalRegion r, boolean isCreate, boolean g
           disableCopyOnRead, preferCD, requestingClient, clientEvent, false);
       trackBucketForTx(key);
       return retVal;
-    } catch (TransactionDataRebalancedException | PrimaryBucketException re) {
-      throw getTransactionException(key, re);
+    } catch (TransactionDataRebalancedException transactionDataRebalancedException) {
```

Comment:
Would it make sense to pull all this handling into a common method?  I see the negatives of doing this as well but it would make updating this code in the future a lot easier

Reply:
The reason to catch separate exceptions is to avoid instance checks. l think it is preferable to avoid instance check in the code.

Reply:
Refactoring partial codes.